test/FastTests/Client/RavenQueryTests.cs

```diff
@@ -0,0 +1,166 @@
+                        var returnedEmployee12 = returnedEmployees1.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' select FirstName, LastName, id() as EmployeeId",
+                            queryUnnamedClassString);
+                        Assert.Equal(2, returnedEmployees1.Count);
+
+                        Assert.Equal(employee1.Id, returnedEmployee11.EmployeeId);
+                        Assert.Equal(employee1.FirstName, returnedEmployee11.FirstName);
+                        Assert.Equal(employee1.LastName, returnedEmployee11.LastName);
+                        Assert.Equal(employee2.Id, returnedEmployee12.EmployeeId);
+
+
+                        //Test 2:
+                        var queryUnnamedClass2 = session.Query<Employees_ByFirstName.IndexEntry, Employees_ByFirstName>()
+                            .Select(a => new { FullName = a.FirstName + " " + a.LastName, EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a) });
+
+                        var queryUnnamedClass2String = queryUnnamedClass2.ToString();
+                        var returnedEmployees2 = queryUnnamedClass2.ToList();
+                        var returnedEmployee21 = returnedEmployees2.First();
+                        var returnedEmployee22 = returnedEmployees2.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' as a select { FullName " +
+                            ": a.FirstName+\" \"+a.LastName, EmployeeId : id(a) }",
+                            queryUnnamedClass2String);
+                        Assert.Equal(2, returnedEmployees2.Count);
+
+                        Assert.Equal(employee1.FirstName + " " + employee1.LastName, returnedEmployee21.FullName);
+                        Assert.Equal(employee1.Id, returnedEmployee21.EmployeeId);
+                        Assert.Equal(employee2.FirstName + " " + employee2.LastName, returnedEmployee22.FullName);
+                        Assert.Equal(employee2.Id, returnedEmployee22.EmployeeId);
+
+
+                        //Test 3:
+                        var queryNamedClass = session.Query<Employees_ByFirstName.IndexEntry, Employees_ByFirstName>()
+                            .Select(a => new EmployeeProjection
+                            {
+                                FullName = a.FirstName + " " + a.LastName, EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a)
+                            });
+
+                        var queryNamedClassString = queryNamedClass.ToString();
+                        var returnedEmployees3 = queryNamedClass.ToList();
+                        var returnedEmployee31 = returnedEmployees3.First();
+                        var returnedEmployee32 = returnedEmployees3.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' as a select { FullName : " +
```

Comment:
can we put it in 1 line without concatenation?

Reply:
Done.